docs/utilities/config_annotations.md

```diff
@@ -0,0 +1,234 @@
+
+    This behavior is partly an oversight and partly a technical limitation. Forge has no intention of punishing any modder who conforms to standard practices. Ideally future releases of Forge will automatically convert traditionally-named variables into conforming property names by replacing capital characters with respective lowercase characters preceded by the underscore character `_`.
+
+!! Note
+
+    The `@Name` annotation cannot currently be applied to a inner class grouping, an oversight which is likely to change in a future version of Forge.
+
+Internationalization
+-------------------------
+
+Since the annotation-based configuration system ties into the in-game mod configuration menu, it becomes important to consider the need to display comments, property names, and category names in the native language of the end user. For this reason the `@LangKey` annotation exists. This annotation is similar to the `@Name` annotation, however, instead of changing the property name, it changes the key that is used for localization purposes. By default, without this annotation, Forge will attempt to localize a property or category for in-game display by using the element's fully-qualified name as the language key.
+
+An example usage of the `@LangKey` is as follows:
+
+````Java
+@Config(modid = "my_mod_id")
+public class MyModConfig {
+
+  @Comment("This property has a non-default language key.")
+  @LangKey("my_mod_id.some_other_category.my_international_property")
+  public static boolean myInternationalProperty = false;
+
+  @LangKey("my_mod_id.some_other_category")
+  public static class MyCategory {
+
+      /* ... */
+
+  }
+
+}
+````
+
+!! Important
+
+    Applying this annotation to a category does not alter the language key of properties contained in that category.
+
+
+Time Specific Properties
+-------------------------
+
+Occasionally in order for a property change to take effect, a mod may require either a complete restart of Minecraft/Forge or a restart of the world. This distinction is important, as a property may control whether or not a sub-module is loaded or not, and a sudden change in the value of the property might cause the system to go looking for an unloaded sub-module. Ideally, the sub-module could be loaded on-the-fly, but in the case of a sub-module that adds a block to the game, attempting to load the sub-module would cause an error as blocks can only be added during initialization phases. Consequently, there exists a need as to signal the configuration system as to hold off making a change to the value specific properties in a `@Config` class until either a world reload occurs or a game reload occurs. For this reason, the annotations `@RequiresWorldRestart` and `@RequiresMcRestart` exist.
+
+The `@RequiresWorldRestart` annotation works by holding off applied changes to a configuration file until a world reload occurs, at which point the changes are then applied, and the world reloaded.
```

Comment:
Does the world reload before or after the change is made? By this sentence, the sequence would be: restart, change value, restart again. I highly doubt that's what is actually done.